app/src/main/java/org/jellyfin/androidtv/ui/presentation/CardPresenter.java

```diff
@@ -410,7 +421,29 @@ public void onBindViewHolder(Presenter.ViewHolder viewHolder, Object item) {
             }
         }
 
-        holder.updateCardViewImage(rowItem.getImageUrl(holder.mCardView.getContext(), mImageType, holder.getCardHeight()));
+        BitmapDrawable blurHashDrawable = null;
+        if (rowItem.getBaseItem() != null && rowItem.getBaseItem().getImageBlurHashes() != null) {
+            HashMap<String, String> blurHashMap;
+            String imageTag;
+            if (aspect == ASPECT_RATIO_BANNER) {
+                blurHashMap = rowItem.getBaseItem().getImageBlurHashes().get(org.jellyfin.apiclient.model.entities.ImageType.Banner);
+                imageTag = rowItem.getBaseItem().getImageTags().get(org.jellyfin.apiclient.model.entities.ImageType.Banner);
+            } else if (aspect == ImageUtils.ASPECT_RATIO_16_9 && !isUserView) {
+                blurHashMap = rowItem.getBaseItem().getImageBlurHashes().get(org.jellyfin.apiclient.model.entities.ImageType.Thumb);
+                imageTag = (rowItem.getPreferParentThumb()) ? rowItem.getBaseItem().getParentThumbImageTag() : rowItem.getBaseItem().getImageTags().get(org.jellyfin.apiclient.model.entities.ImageType.Thumb);
+            } else {
+                blurHashMap = rowItem.getBaseItem().getImageBlurHashes().get(org.jellyfin.apiclient.model.entities.ImageType.Primary);
+                imageTag = rowItem.getBaseItem().getImageTags().get(org.jellyfin.apiclient.model.entities.ImageType.Primary);
+            }
+
+            if (blurHashMap != null && !blurHashMap.isEmpty() && imageTag != null) {
+                String blurHash = blurHashMap.get(imageTag);
+                Bitmap blurHashBitmap = BlurHashDecoder.INSTANCE.decode(blurHash, 32, 32, 1, true);
```

Comment:
I'm wondering if creating a bitmap with the correct aspect ratio would make more sense. Right now you're creating a 32x32 bitmap but if you multiply the width with the aspect ratio it would be proportional to the actual image.

Reply:
Definitely looks better using the correct aspect ratio. I pushed a commit that keeps one dimension at 32 and adjusts the other by the aspect ratio. I'm not sure if there's potential for a performance hit with banner images since they end up being 173x32. Posters are 32x48, and thumbs are 57x32. I can flip it so 32 is the maximum for a dimension if you think performance might be an issue.

Reply:
I think it's fine, the decoding shouldn't impact performance that much